File: backend/src/infrastructure/database/models/ProfessorModel.ts
import { Schema, model, Types } from 'mongoose';

export interface PricingConfig {
  individualClass?: number;
  groupClass?: number;
  courtRental?: number;
}

export interface ProfessorDocument {
  _id: Types.ObjectId;
  authUserId: Types.ObjectId; // Reference to AuthUser
  name: string;
  email: string;
  phone: string;
  specialties: string[];
  hourlyRate: number;
  experienceYears: number;
  pricing?: PricingConfig; // Custom pricing (overrides base pricing)
}

const ProfessorSchema = new Schema<ProfessorDocument>(
  {
    authUserId: { type: Schema.Types.ObjectId, ref: 'AuthUser', required: true, unique: true },
    name: { type: String, required: true },
    email: { type: String, required: true, unique: true, index: true },
    phone: { type: String, required: true },
    specialties: { type: [String], default: [] },
    hourlyRate: { type: Number, required: true },
    experienceYears: { type: Number, required: true, default: 0 },
    pricing: {
      individualClass: { type: Number },
      groupClass: { type: Number },
      courtRental: { type: Number },
    },
  },
  { timestamps: true },
);

export const ProfessorModel = model<ProfessorDocument>('Professor', ProfessorSchema);

File: mobile/lib/features/booking/domain/models/professor_model.dart
class PricingConfig {
  final double individualClass;
  final double groupClass;
  final double courtRental;

  PricingConfig({
    required this.individualClass,
    required this.groupClass,
    required this.courtRental,
  });

  factory PricingConfig.fromJson(Map<String, dynamic> json) {
    return PricingConfig(
      individualClass: (json['individualClass'] as num).toDouble(),
      groupClass: (json['groupClass'] as num).toDouble(),
      courtRental: (json['courtRental'] as num).toDouble(),
    );
  }
}

class ProfessorBookingModel {
  final String id;
  final String name;
  final String email;
  final String phone;
  final List<String> specialties;
  final double hourlyRate;
  final PricingConfig pricing;
  final int experienceYears;
  final double rating;

  ProfessorBookingModel({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    required this.specialties,
    required this.hourlyRate,
    required this.pricing,
    required this.experienceYears,
    required this.rating,
  });

  factory ProfessorBookingModel.fromJson(Map<String, dynamic> json) {
    return ProfessorBookingModel(
      id: json['id'] as String,
      name: json['name'] as String,
      email: json['email'] as String,
      phone: json['phone'] as String? ?? '',
      specialties:
          (json['specialties'] as List<dynamic>?)
              ?.map((e) => e as String)
              .toList() ??
          [],
      hourlyRate: (json['hourlyRate'] as num?)?.toDouble() ?? 0.0,
      pricing: PricingConfig.fromJson(json['pricing'] as Map<String, dynamic>),
      experienceYears: (json['experienceYears'] as num?)?.toInt() ?? 0,
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'phone': phone,
      'specialties': specialties,
      'hourlyRate': hourlyRate,
      'experienceYears': experienceYears,
      'rating': rating,
    };
  }
}
